File: JavaScript/greeter.js
'use strict'

function salutation(hour) {
  if (hour >= 6 && hour <= 11) return "Good morning ";
  if (hour >= 18 && hour <= 21) return "Good evening ";
  if (hour >= 22 || hour <= 5) return "Good night ";
  return 'Hello ';
}

module.exports = function greeter(name, time) {
  const formattedName = name.trim()[0].toUpperCase() + name.trim().slice(1);
  const defaultHour = 12;

  const hour = time ? time.getHours() : defaultHour;

  return salutation(hour) + formattedName;
};

File: Ruby/app/greeter.rb
class Greeter

  def greet(name, time = nil)
    formatted_name = name.strip.capitalize

    if time && time >= Time.parse('6am') && time <= Time.parse('12pm')
      return "Good morning #{formatted_name}"
    elsif time && time >= Time.parse('6pm') && time <= Time.parse('10pm')
      return "Good evening #{formatted_name}"
    end

    "Hello #{formatted_name}"
  end
end
